dbt/adapters/duckdb/impl.py

```diff
@@ -176,6 +187,30 @@ def get_rows_different_sql(
         )
         return sql
 
+    @available.parse(lambda *a, **k: [])
+    def get_column_schema_from_query(self, sql: str) -> List[Column]:
+        """Get a list of the Columns with names and data types from the given sql."""
+
+        # Taking advantage of yet another amazing DuckDB SQL feature right here: the
+        # ability to DESCRIBE a query instead of a relation
+        describe_sql = f"DESCRIBE ({sql})"
```

Comment:
@jtcohen6 after a truly circuitous journey, this ended up being simple and delightful to implement.

Reply:
whoa! this is very handy

Reply:
this is pretty sweet